src/hooks/use-document-sync-to-firebase.ts

```diff
@@ -75,6 +75,19 @@ export function useDocumentSyncToFirebase(
 
   const commonSyncEnabled = !disableFirebaseSync && contentStatus === ContentStatus.Valid;
 
+  const syncFirestoreDocumentProp = (prop: string, value?: string) => {
+    const promises = [];
+    const query = firestore.collection("documents").where("key", "==", document.key);
+
+    promises.push(query.get().then((querySnapshot) => {
+      return Promise.all(
+        querySnapshot.docs.map((doc) => doc.ref.update({ [prop]: value}))
+      );
+    }));
+
+    return Promise.all(promises);
```

Comment:
I think the use of `promises`, and `Promise.all(promises)` is unnecessary. There is just one promise returned by `query.get()`. So I think this could be simplified to be just:
```
return query.get().then((querySnapshot) => {
...
```

Or you could switch to the async/await approach and do:
```
const querySnapshot = await query.get();
return Promise.all(
  querySnapshot.docs.map((doc) => doc.ref.update({ [prop]: value}))
);
```

Reply:
You're right, thanks. I've made this change and things seem to work fine.